test/support/runner.rb

```diff
@@ -0,0 +1,48 @@
+module RegexpParserTest
+  class Runner
+    def initialize(arguments, warning_whitelist)
+      @arguments = arguments
+      @warning_whitelist = warning_whitelist
+    end
+
+    def run
+      test_status = nil
+
+      warnings = WarningFilter.use do
+        setup
+        test_status = run_test_unit
+      end
+
+      return false unless test_status
+
+      unexpected = filter_warnings(warnings, warning_whitelist)
+      fail "Unexpected warnings: #{unexpected.to_a.join("\n")}" if unexpected.any?
+    end
+
+    private
+
+    def filter_warnings(warnings, whitelist)
+      warnings.reject do |warning|
+        whitelist.any? { |entry| warning.end_with?(entry) }
+      end
+    end
+
+    def setup
+      $VERBOSE = true
+
+      test_files.each(&method(:require))
+    end
+
+    def run_test_unit
+      Test::Unit::AutoRunner.run
+    end
+
+    def test_files
+      arguments
+        .select { |path| File.exist?(path) && File.file?(path) }
+        .map { |path| File.expand_path(path) }
```

Comment:
Yeah I strongly prefer `Pathname` but I didn't see it used anywhere else in this codebase so I didn't want to impose too much of my style with this contribution